### PR TITLE
rolling up goals and objectives

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Roll up goals & objectives in the UI
- The RTR table should roll up goals and objectives with matching text/status, and combine objectives under goals and ARs under objectives
- The goal picker should combine goals, and display all matching goal numbers

## How to test
See description

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-898

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/components/GoalsTable/ObjectiveRow.js [hunk 1]
```diff
@@ -112,16 +106,25 @@ function ObjectiveRow({
...
+                  </Link>
+                </li>
+              );
+            })}
+          </ul>
         </li>
         <li className="padding-x-105 padding-y-0">
           <span className="sr-only">Grant number </span>
-          {grantNumber}
+          {grantNumbers.join(' ')}
```
Comment (GarrettEHill) [inline frontend/src/components/GoalsTable/ObjectiveRow.js:127]: Should this be ', '

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good.

I was able to use existing goals and see the new objectives appear under the goal. 

The only issue I noticed was the '???' for the region of the AR.

**Comment by jasalisbury** [comment]
https://user-images.githubusercontent.com/6631790/176198085-4924bf86-5a99-4a57-9037-8cf0296f0904.mp4

Goal count seems to be messing with the pagination. Also do we need to roll up the goal status widget at the top of the page as well?

**Comment by jasalisbury** [comment]
Also does this handle changing the status of a "rolled up" goal?

File: frontend/src/components/GoalsTable/__tests__/GoalsTable.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ const goals = [{
   goalText: 'This is goal text 1.',
   goalTopics: ['Human Resources', 'Safety Practices', 'Program Planning and Services'],
   objectiveCount: 5,
-  goalNumber: 'R14-G-4598',
+  goalNumbers: ['R14-G-4598'],
```
[the remark this answers is not in the document]
Comment (jasalisbury) [reply frontend/src/components/GoalsTable/__tests__/GoalsTable.js:45]: They don't anymore, they should be just `G-${id}` now (not sure where that change is or if it has made it in yet)

File: frontend/src/pages/RecipientRecord/pages/PrintGoals.js [hunk 3]
```diff
@@ -25,7 +26,7 @@ export default function PrintGoals({ location, recipientId, regionId }) {
           regionId,
           sortConfig.sortBy,
           sortConfig.direction,
-          sortConfig.offset,
+          OFFSET,
```
Comment (jasalisbury) [inline frontend/src/pages/RecipientRecord/pages/PrintGoals.js:29]: This always being zero is intentional?

Comment (thewatermethod) [reply frontend/src/pages/RecipientRecord/pages/PrintGoals.js:29]: yeah, this function will be outta here when you are able to select specific goals

**Pull request merged**